lib/class/art.class.php

```diff
@@ -784,8 +784,17 @@ public function generate_thumb($image, $size, $mime)
      * @param string $type
      * @return string|null
      */
-    public static function get_from_source($data, $type = 'album')
+    public static function get_from_source($data, $type)
     {
+        if (!isset($type)) {
+            $show_song_art = AmpConfig::get('show_song_art') ?: false;
+            if ($show_song_art) {
```

Comment:
Uhhhhhhhh that's... uhhhhhh

This basically means:
If there's ANYTHING else than `0; "0"; ""; null` in the variable, we get `$type = 'song';`.
Even if we would have `'false'` in it, it would be true.

please take a look at this: https://www.php.net/manual/en/types.comparisons.php

We need a stricter comparison here. :)